includes/api/coupon-get.php

```diff
@@ -67,6 +67,14 @@ function getCouponList($request)
         'posts_per_page' => -1, // By default WP_Query will return only 10 posts, to avoid that we need to pass -1
     );
 
+    //check woo-discount-rule plugin disabling the coupons
+    $discountOptions = get_option( 'woo-discount-config-v2', array());
+    $isCouponEnabled = $discountOptions['disable_coupon_when_rule_applied'];
```

Comment:
Incase of `get_option( 'woo-discount-config-v2', array());` fails or empty array, i guess it will throw an error here @poojarazor

Reply:
Error handling is missing.

Reply:
Incase of empty array it will not throw any error, it will set as a  null value

Reply:
@poojarazor Can user apply the coupon manually?